Added reddit icon and OverlayShare component

Dishebh opened the pull request:
Fixes issue: #791 

Preview Link: https://deploy-preview-845--peaceful-nightingale-55a1b7.netlify.app/interface/overlayShare

Marvin9 commented on src/ignitus-Shared/ignitus-DesignSystem/ignitus-Molecules/ignitus-OverlayShare/Components/index.tsx:
```diff
@@ -0,0 +1,58 @@
+    </O.TopSection>
+    <O.BottomSection>
+      <O.Paragraph>{Link}</O.Paragraph>
+      <DefaultIconInput
```
Input should be readonly.

Dishebh commented on src/ignitus-Shared/ignitus-DesignSystem/ignitus-Molecules/ignitus-OverlayShare/Components/index.tsx:
```diff
@@ -0,0 +1,58 @@
+        name={AppIcon.CopyIcon}
+        type="string"
+        placeholder="Paste URL Here"
+        handleClick={copy}
```
For this, I have to make changes to the `DefaultIconInput` component first. This becomes an issue for a separate component, I guess.

Marvin9 replied:
Sorry @Dishebh , I didn't notice this reply. I will figure it out and get back to you.

Marvin9 replied:
I'll update in  next PR. Keep it as it is. ```ref``` is what we can  pass to input but It will be too much in one PR.

Dishebh replied:
Thanks @Marvin9 :)

Paarmita commented on src/ignitus-Shared/ignitus-DesignSystem/ignitus-Molecules/ignitus-OverlayShare/styles.ts:
```diff
@@ -0,0 +1,56 @@
+  font-size: ${MD};
+  margin: 0.6rem;
+`;
+
```
Add some space between the 2 sections.
<img width="314" alt="Screenshot 2020-05-24 at 11 59 47 AM" src="https://user-images.githubusercontent.com/26546816/82751229-1c24bd80-9db6-11ea-9aaa-4e06fdcbceea.png">

Dishebh replied:
Added some extra space.

Paarmita commented on src/ignitus-UserInterfaceBook/InterfaceSideNavigation/constants.ts:
```diff
@@ -125,6 +125,10 @@ export const allEdges: Edges[] = [
       },
+      {
+        title: 'Overlay Share',
+        route: '/interface/overlayShare',
```
Create a parent `Overlay` under `Molecules` and add the Hooray and share overlays as children.

Dishebh replied:
Bifurcated into children

Dishebh commented on src/ignitus-Shared/ignitus-DesignSystem/ignitus-Molecules/ignitus-OverlayShare/styles.ts:
```diff
@@ -0,0 +1,56 @@
+`;
+
+export const TopSection = styled(flexibleRowDiv)`
+  padding: 2rem;
```
Done

Dishebh commented on src/ignitus-Shared/ignitus-DesignSystem/ignitus-Molecules/ignitus-OverlayShare/styles.ts:
```diff
@@ -0,0 +1,56 @@
+  cursor: pointer;
+  overflow: hidden;
+  margin: 0 0.2rem;
+  height: 2.8rem;
```
The Facebook icon still seems a little off. I tried changing the padding / margin but then it made other icons off their center position. Please have a look.

Dishebh commented:
@Paarmita please see if the UI changes are fine :)

Dishebh commented:
The merge conflicts arise as I have deleted those files. They will go once merged. Thanks!

divyanshu-rawat commented:
but we can't merge until the are resolved @Dishebh.

Dishebh commented:
@divyanshu-rawat, as @Paarmita said, I have changed the architecture to:

```
├── Molecules
|   ├── Overlay 
|   |   ├── Overlay Share
|   |   └── Overlay Hooray
```

and so, I renamed the `Overlay` file to `Overlay Hooray`, then migrated (copy and pasted) the files to maintain the layout shown above.

Dishebh commented:
@divyanshu-rawat I see that the `Overlay` has been renamed to `Modal`, and so that's why the checks fail. So, shall I make the changes accordingly in this PR,  and change the names to `Modal`, `Modal Hooray` and `Modal Share` (from `Overlay`) as well?

divyanshu-rawat commented:
Sounds good @Dishebh :)

divyanshu-rawat commented:
It would be great if you create a category Modal and then put `Modal Hooray `& `Modal Share `as a sub-menu item in it in the side menu, on expanding modal, `Modal Hooray `& `Modal Share should pop up.

Dishebh commented:
@divyanshu-rawat sure thing :+1: It would be great if you could review my other PRs as well (sorry for pinging you again, I know the review process takes time but the gssoc program ends tomorrow, so would really appreciate it if you could do so). 

Thanks a lot for your guidance :)

divyanshu-rawat commented:
@Dishebh I will try my best :)

Dishebh commented:
@divyanshu-rawat all checks passed now :tada:

divyanshu-rawat commented:
Thanks @Dishebh :)
